sdk/go/pulumi-language-go/main.go

```diff
@@ -395,7 +414,9 @@ func (host *goLanguageHost) Run(ctx context.Context, req *pulumirpc.RunRequest)
 		}
 
 		cmd := exec.Command(gobin, "run", cwd)
-		if err := execProgramCmd(cmd, env); err != nil {
+
+		status, err := runCmdStatus(cmd, env)
```

Comment:
This looks right but would benefit from a comment that we use custom err handling here because go run mangles exit codes right.. So we don't expect e.g. 213 code from go run.